tests/e2e/bulker.test.ts

```diff
@@ -0,0 +1,1021 @@
+          for (const signature of bulker.signatures$.getValue()) {
+            // @ts-ignore
+            await bulker.sign(signature);
+          }
+          await bulker.executeBatch();
+
+          const ma3Balance = await morphoAaveV3.borrowBalance(
+            weth.address,
+            morphoUser.address
+          );
+          expect(ma3Balance).to.equal(
+            constants.Zero,
+            "ma3 balance should be 0"
+          );
+
+          expect(await weth.balanceOf(addresses.bulker)).to.be.equal(
+            constants.Zero,
+            "weth balance left is not 0"
+          );
+
+          const wethBalance = await weth.balanceOf(morphoUser.address);
+          expect(wethBalance).to.be.greaterThan(
+            initialWethBalance.sub(interests),
+            `weth balance (${wethBalance}) is not initialWethBalance (${initialWethBalance})`
+          );
+        });
+      });
+
+      it("should repay WETH with ETH", async () => {
+        await approve(contractAddress, async () => {
+          const oneEth = utils.parseEther("1");
+          await morphoAaveV3.supplyCollateral(
+            Underlying.dai,
+            initialDaiBalance,
+            morphoUser.address
+          );
+          await morphoAaveV3.borrow(
+            Underlying.weth,
+            oneEth,
+            morphoUser.address,
+            morphoUser.address,
+            10
+          );
+          await weth.withdraw(await weth.balanceOf(morphoUser.address));
+          await morphoAdapter.refreshAll("latest");
+          const oldBalance = await morphoUser.getBalance();
+          expect(await weth.balanceOf(morphoUser.address)).to.equal(
+            constants.Zero,
+            "weth balance should be 0"
+          );
+          await bulker.addOperations([
+            {
+              type: TransactionType.repay,
+              amount: constants.MaxUint256,
+              underlyingAddress: Underlying.weth,
+            },
+          ]);
+
+          for (const signature of bulker.signatures$.getValue()) {
+            // @ts-ignore
+            await bulker.sign(signature);
+          }
+          await bulker.executeBatch();
+
+          const ma3Balance = await morphoAaveV3.borrowBalance(
+            weth.address,
+            morphoUser.address
+          );
+          expect(ma3Balance).to.equal(
+            constants.Zero,
+            "ma3 balance should be 0"
+          );
+
+          expect(await weth.balanceOf(addresses.bulker)).to.be.equal(
+            constants.Zero,
+            "weth balance left is not 0"
+          );
+
+          const wethBalance = await weth.balanceOf(morphoUser.address);
+          // TODO I don't know how to solve this test, there's always remaining weth in user wallet
+          // expect(wethBalance).to.be.lessThan(
+          //   dust,
+          //   "weth balance should be unchanged (modulo dust)"
+          // );
+
+          // Expect oldBalance - 2 ETH < finalAmount < oldBalance - 1 ETH
```

Comment:
it's normal to have remaining weth, in order to repay max, you need to wrap a bit more of eth than the user's position so that the bulker contract have enough to repay the position + the interests accumulated during the blockj inclusion latency. Then the bulker contract sends back the excess of wrapped ETH

Reply:
you have to check that the remaining amount is not greater than 0.01% of the borrow position ([that's the buffer we add](https://github.com/morpho-org/morpho-aave-v3-sdk/blob/12bff66972c0900a2b969333f1b5b2cd8cfe99eb/src/txHandler/Bulker.TxHandler.ts#L730-L753))